### PR TITLE
Use UserConnection to keep track of User to User connections

### PR DESCRIPTION
**Pull request by gmethvin**
Don't merge yet. This has some significant changes to the social graph so we need to be careful.

### CODE REVIEW

File: shoebox/conf/evolutions/shoebox/57.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    state varchar(20) NOT NULL,
+    created_at datetime NOT NULL,
+    updated_at datetime NOT NULL,
+
+    PRIMARY KEY (id),
+
+    CONSTRAINT user_connection_user_1 FOREIGN KEY (user_1) REFERENCES user(id),
+    CONSTRAINT user_connection_user_2 FOREIGN KEY (user_2) REFERENCES user(id),
+    UNIQUE INDEX(user_1, user_2)
```
Comment (eishay) [inline shoebox/conf/evolutions/shoebox/57.sql:16]: maybe 

``` sql
UNIQUE INDEX(user_1, user_2, state) 
```

?

Comment (gmethvin) [inline shoebox/conf/evolutions/shoebox/57.sql:16]: I think we often want to find the relationship between user_1 and user_2 regardless of state (so we can activate it if it's inactive). This is the same thing we're doing in social_connection.

Comment (eishay) [inline shoebox/conf/evolutions/shoebox/57.sql:16]: ok

File: shoebox/test/com/keepit/model/CommentTest.scala [hunk 2]
```diff
@@ -80,12 +71,7 @@ class CommentTest extends Specification with DbRepos {
           commentRepo.getLastPublicIdByConnection(user1.id.get, uri1.id.get).isEmpty === true
           commentRepo.getLastPublicIdByConnection(user2.id.get, uri1.id.get).isEmpty === true
           commentRepo.getLastPublicIdByConnection(user3.id.get, uri1.id.get).isEmpty === true
-
-          val socialUserInfoRepo = inject[SocialUserInfoRepo]
-          val socialUser1 = socialUserInfoRepo.save(SocialUserInfo(userId = user1.id, fullName = "Andrew Conner", socialId = SocialId("1111111"), networkType = SocialNetworks.FACEBOOK))
-          val socialUser2 = socialUserInfoRepo.save(SocialUserInfo(userId = user2.id, fullName = "Eishay Smith", socialId = SocialId("2222222"), networkType = SocialNetworks.FACEBOOK))
-          val socialConnRepo = inject[SocialConnectionRepo]
-          socialConnRepo.save(SocialConnection(socialUser1 = socialUser1.id.get, socialUser2 = socialUser2.id.get))
+          userConnRepo.save(UserConnection(user1 = user1.id.get, user2 = user2.id.get))
```
Comment (2is10) [inline shoebox/test/com/keepit/model/CommentTest.scala:74]: hooray for less test boilerplate!

**Pull request merged**